agent.go

```diff
@@ -985,9 +987,30 @@ func (a *Agent) sendBindingSuccess(m *stun.Message, local, remote Candidate) {
 	}
 }
 
+// Removes pending binding requests that are over maxBindingRequestTimeout old
+func (a *Agent) invalidatePendingBindingRequests() int {
+	nRemoved := 0
+	temp := a.pendingBindingRequests[:0]
+	for _, bindingRequest := range a.pendingBindingRequests {
+		if time.Since(bindingRequest.timestamp) < maxBindingRequestTimeout {
+			temp = append(temp, bindingRequest)
+		} else {
+			nRemoved++
+		}
+	}
+	// Garbage collect removed elements
+	for i := len(temp); i < len(a.pendingBindingRequests); i++ {
+		a.pendingBindingRequests[i] = bindingRequest{}
+	}
```

Comment:
Will GC never happen if we do this? Is this a language or implementation detail? 

We don't do this in other places so I would push back against this (unless you measured and it is a real problem!)